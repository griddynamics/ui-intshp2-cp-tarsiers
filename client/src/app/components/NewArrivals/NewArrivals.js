import React from 'react';
import PropTypes from 'prop-types';

import Carousel from '../../shared/Carousel';
import ProductItemContainer from '../../shared/ProductItem/ProductItemContainer';
import {
  isAddedToCart,
  isAddedToWishList
} from '../../../utils/inCartInWishlist.service';
import styles from './NewArrivals.module.scss';

const title = 'New Arrivals';

const NewArrivals = props => {
  const { newArrivals, wishlist, cart } = props;
  const extended = true;
  const titleArr = title.split(' ');

  const list =
    newArrivals &&
    newArrivals.map(el => (
      <ProductItemContainer
        isAddedToWishList={isAddedToWishList(el._id, wishlist)}
        isAddedToCart={isAddedToCart(el._id, cart)}
        extended={extended}
        key={el._id}
        data={el}
      />
    ));

  return (
    <section className={styles.arrivals}>
      <div className="section_heading">
        <h2>
          <span className="highlighted">{titleArr[0]}</span>
          &nbsp;
          {titleArr[1]}
        </h2>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry
        </p>
      </div>
      <div className={styles.arrivals__display}>
        <Carousel data={newArrivals} extended>
          {list}
        </Carousel>
      </div>
    </section>
  );
};

NewArrivals.propTypes = {
  newArrivals: PropTypes.arrayOf(PropTypes.object)
};

NewArrivals.defaultProps = {
  newArrivals: []
};

export default NewArrivals;
