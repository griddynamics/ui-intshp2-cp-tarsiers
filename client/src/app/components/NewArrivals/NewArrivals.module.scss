@import '../../../styles/variables';
@import '../../../styles/mixin';

.arrivals {
  background-color: $color-light-1;
  color: $color-dark;
  padding: 80px 0;
  max-width: $media-large;
  margin: 0 auto;
  text-align: center;

  &__display {
    margin: 0 auto;
    text-align: center;
    @include d-flex() {
      align-items: center;
    }
  }
}
