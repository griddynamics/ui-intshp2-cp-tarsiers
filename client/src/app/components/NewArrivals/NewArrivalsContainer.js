import { connect } from 'react-redux';
import NewArrivals from './NewArrivals';

const mapStateToProps = state => ({
  newArrivals: state.products.newArrivals,
  wishlist: state.wishlist.wishlist,
  cart: state.cart
});

const NewArrivalsContainer = connect(mapStateToProps)(NewArrivals);

export default NewArrivalsContainer;
