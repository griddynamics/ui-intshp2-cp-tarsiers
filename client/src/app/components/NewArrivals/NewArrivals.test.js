import React from 'react';
import { shallow } from 'enzyme';
import NewArrivals from './NewArrivals';

describe('<NewArrivals />', () => {
  it('should match its snapshot', () => {
    const props = { products: [], extended: true };
    const wrapper = shallow(<NewArrivals {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
