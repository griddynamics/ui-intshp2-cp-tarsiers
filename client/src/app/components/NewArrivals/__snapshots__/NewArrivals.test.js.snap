// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewArrivals /> should match its snapshot 1`] = `
<section
  className="arrivals"
>
  <div
    className="section_heading"
  >
    <h2>
      <span
        className="highlighted"
      >
        New
      </span>
      Â 
      Arrivals
    </h2>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry
    </p>
  </div>
  <div
    className="arrivals__display"
  >
    <Carousel
      data={Array []}
      extended={true}
    />
  </div>
</section>
`;
